br/pkg/storage/helper.go

```diff
@@ -48,22 +52,32 @@ func UnmarshalDir[T any](ctx context.Context, walkOpt *WalkOption, s ExternalSto
 	errCh := make(chan error, 1)
 	reader := func() {
 		defer close(ch)
+		pool := util.NewWorkerPool(128, "metadata")
+		eg, ectx := errgroup.WithContext(ctx)
 		err := s.WalkDir(ctx, walkOpt, func(path string, size int64) error {
-			metaBytes, err := s.ReadFile(ctx, path)
```

Comment:
check whether `ectx` is canceled here because the error is occurred in `ApplyOnErrorGroup` won't stop `WalkDir`.